In [2]:
import numpy as np
import pickle, os, torch
import matplotlib.pyplot as plt
from torchvision import datasets, transforms
import models

os.environ['KMP_DUPLICATE_LIB_OK']='True'



In [32]:
Exp = "F2C_1e3-1e2_b8_e50_d_no-reset"
Exp_desc = "Freezing 2 first convs - lr=1e-3 -> 1e-2 - batch_ratio=8 - pre_train_epochs=50 - using dropout - no-reset"


path = os.path.join("experiments_results", "train_fewshot", Exp)

with open(os.path.join(path, "ann_logs.pickle"), "rb") as f:
    ann_logs = pickle.load(f)
    
with open(os.path.join(path, "snn_logs.pickle"), "rb") as f:
    snn_logs = pickle.load(f)

print("## {}".format(Exp_desc))
print("|     | Labels | Pre-training on [0-6] | 1 shot | 5 shot | 10 shot | 20 shot | 50 shot |")
print("| :-: | :----: | :-------------------: | :----: | :----: | :-----: | :-----: | :-----: |")
print("| ANN | [0-6]  | {:0.2f}% / {:0.2f}%   |{:0.2f}%|{:0.2f}%|{:0.2f}% |{:0.2f}% |{:0.2f}% |".format(max(ann_logs["pre-train_acc"]),max(ann_logs["pre-test_acc"]), ann_logs["1-shot_pre-test_acc"][0], ann_logs["5-shot_pre-test_acc"][0], ann_logs["10-shot_pre-test_acc"][0], ann_logs["20-shot_pre-test_acc"][0], ann_logs["50-shot_pre-test_acc"][0]))
print("|     | [7-9]  | -                     |{:0.2f}%|{:0.2f}%|{:0.2f}% |{:0.2f}% |{:0.2f}% |".format(ann_logs["1-shot_test_acc"][0], ann_logs["5-shot_test_acc"][0], ann_logs["10-shot_test_acc"][0], ann_logs["20-shot_test_acc"][0], ann_logs["50-shot_test_acc"][0]))
print("| SNN | [0-6]  | {:0.2f}% / {:0.2f}%   |{:0.2f}%|{:0.2f}%|{:0.2f}% |{:0.2f}% |{:0.2f}% |".format(max(snn_logs["pre-train_acc"]),max(snn_logs["pre-test_acc"]), snn_logs["1-shot_pre-test_acc"][0], snn_logs["5-shot_pre-test_acc"][0], snn_logs["10-shot_pre-test_acc"][0], snn_logs["20-shot_pre-test_acc"][0], snn_logs["50-shot_pre-test_acc"][0]))
print("|     | [7-9]  | -                     |{:0.2f}%|{:0.2f}%|{:0.2f}% |{:0.2f}% |{:0.2f}% |".format(snn_logs["1-shot_test_acc"][0], snn_logs["5-shot_test_acc"][0], snn_logs["10-shot_test_acc"][0], snn_logs["20-shot_test_acc"][0], snn_logs["50-shot_test_acc"][0]))


# Freezing 2 first convs - lr=1e-3 -> 1e-2 - batch_ratio=8 - pre_train_epochs=50 - using dropout - no-reset
|     | Labels | Pre-training on [0-6] | 1 shot | 5 shot | 10 shot | 20 shot | 50 shot |
| :-: | :----: | :-------------------: | :----: | :----: | :-----: | :-----: | :-----: |
| ANN | [0-6]  | 99.75% / 99.64%   |99.50%|99.43%|98.31% |99.26% |99.10% |
|     | [7-9]  | -                     |0.00%|0.00%|0.00% |0.00% |0.00% |
| SNN | [0-6]  | 99.61% / 99.61%   |99.59%|98.44%|96.19% |96.18% |88.52% |
|     | [7-9]  | -                     |0.00%|80.00%|83.33% |91.67% |86.00% |


## Freezing 2 first convs - lr=1e-3 -> 1e-2 - batch_ratio=8 - pre_train_epochs=50 - using dropout - no-reset
|     | Labels | Pre-training on [0-6] | 1 shot | 5 shot | 10 shot | 20 shot | 50 shot |
| :-: | :----: | :-------------------: | :----: | :----: | :-----: | :-----: | :-----: |
| ANN | [0-6]  | 99.75% / 99.64%   |99.50%|99.43%|98.31% |99.26% |99.10% |
|     | [7-9]  | -                     |0.00%|0.00%|0.00% |0.00% |0.00% |
| SNN | [0-6]  | 99.61% / 99.61%   |99.59%|98.44%|96.19% |96.18% |88.52% |
|     | [7-9]  | -                     |0.00%|80.00%|83.33% |91.67% |86.00% |

In [33]:
Exp = "F2C_1e3-1e2_b8_e50_d_reset"
Exp_desc = "Freezing 2 first convs - lr=1e-3 -> 1e-2 - batch_ratio=8 - pre_train_epochs=50 - using dropout - reset"


path = os.path.join("experiments_results", "train_fewshot", Exp)

with open(os.path.join(path, "ann_logs.pickle"), "rb") as f:
    ann_logs = pickle.load(f)
    
with open(os.path.join(path, "snn_logs.pickle"), "rb") as f:
    snn_logs = pickle.load(f)

print("## {}".format(Exp_desc))
print("|     | Labels | Pre-training on [0-6] | 1 shot | 5 shot | 10 shot | 20 shot | 50 shot |")
print("| :-: | :----: | :-------------------: | :----: | :----: | :-----: | :-----: | :-----: |")
print("| ANN | [0-6]  | {:0.2f}% / {:0.2f}%   |{:0.2f}%|{:0.2f}%|{:0.2f}% |{:0.2f}% |{:0.2f}% |".format(max(ann_logs["pre-train_acc"]),max(ann_logs["pre-test_acc"]), ann_logs["1-shot_pre-test_acc"][0], ann_logs["5-shot_pre-test_acc"][0], ann_logs["10-shot_pre-test_acc"][0], ann_logs["20-shot_pre-test_acc"][0], ann_logs["50-shot_pre-test_acc"][0]))
print("|     | [7-9]  | -                     |{:0.2f}%|{:0.2f}%|{:0.2f}% |{:0.2f}% |{:0.2f}% |".format(ann_logs["1-shot_test_acc"][0], ann_logs["5-shot_test_acc"][0], ann_logs["10-shot_test_acc"][0], ann_logs["20-shot_test_acc"][0], ann_logs["50-shot_test_acc"][0]))
print("| SNN | [0-6]  | {:0.2f}% / {:0.2f}%   |{:0.2f}%|{:0.2f}%|{:0.2f}% |{:0.2f}% |{:0.2f}% |".format(max(snn_logs["pre-train_acc"]),max(snn_logs["pre-test_acc"]), snn_logs["1-shot_pre-test_acc"][0], snn_logs["5-shot_pre-test_acc"][0], snn_logs["10-shot_pre-test_acc"][0], snn_logs["20-shot_pre-test_acc"][0], snn_logs["50-shot_pre-test_acc"][0]))
print("|     | [7-9]  | -                     |{:0.2f}%|{:0.2f}%|{:0.2f}% |{:0.2f}% |{:0.2f}% |".format(snn_logs["1-shot_test_acc"][0], snn_logs["5-shot_test_acc"][0], snn_logs["10-shot_test_acc"][0], snn_logs["20-shot_test_acc"][0], snn_logs["50-shot_test_acc"][0]))


## Freezing 2 first convs - lr=1e-3 -> 1e-2 - batch_ratio=8 - pre_train_epochs=50 - using dropout - reset
|     | Labels | Pre-training on [0-6] | 1 shot | 5 shot | 10 shot | 20 shot | 50 shot |
| :-: | :----: | :-------------------: | :----: | :----: | :-----: | :-----: | :-----: |
| ANN | [0-6]  | 99.75% / 99.64%   |0.29%|0.31%|0.00% |0.00% |0.00% |
|     | [7-9]  | -                     |33.33%|86.67%|76.67% |90.00% |96.00% |
| SNN | [0-6]  | 99.63% / 99.61%   |10.47%|0.33%|0.00% |0.54% |0.14% |
|     | [7-9]  | -                     |0.00%|86.67%|80.00% |98.33% |94.00% |


## Freezing 2 first convs - lr=1e-3 -> 1e-2 - batch_ratio=8 - pre_train_epochs=50 - using dropout - reset
|     | Labels | Pre-training on [0-6] | 1 shot | 5 shot | 10 shot | 20 shot | 50 shot |
| :-: | :----: | :-------------------: | :----: | :----: | :-----: | :-----: | :-----: |
| ANN | [0-6]  | 99.75% / 99.64%   |0.29%|0.31%|0.00% |0.00% |0.00% |
|     | [7-9]  | -                     |33.33%|86.67%|76.67% |90.00% |96.00% |
| SNN | [0-6]  | 99.63% / 99.61%   |10.47%|0.33%|0.00% |0.54% |0.14% |
|     | [7-9]  | -                     |0.00%|86.67%|80.00% |98.33% |94.00% |

In [3]:
Exp = "F2C_1e3-1e2_b8_e50_nod_reset"
Exp_desc = "Freezing 2 first convs - lr=1e-3 -> 1e-2 - batch_ratio=8 - pre_train_epochs=50 - no dropout - reset"


path = os.path.join("experiments_results", "train_fewshot", Exp)

with open(os.path.join(path, "ann_logs.pickle"), "rb") as f:
    ann_logs = pickle.load(f)
    
with open(os.path.join(path, "snn_logs.pickle"), "rb") as f:
    snn_logs = pickle.load(f)

print("## {}".format(Exp_desc))
print("|     | Labels | Pre-training on [0-6] | 1 shot | 5 shot | 10 shot | 20 shot | 50 shot |")
print("| :-: | :----: | :-------------------: | :----: | :----: | :-----: | :-----: | :-----: |")
print("| ANN | [0-6]  | {:0.2f}% / {:0.2f}%   |{:0.2f}%|{:0.2f}%|{:0.2f}% |{:0.2f}% |{:0.2f}% |".format(max(ann_logs["pre-train_acc"]),max(ann_logs["pre-test_acc"]), ann_logs["1-shot_pre-test_acc"][0], ann_logs["5-shot_pre-test_acc"][0], ann_logs["10-shot_pre-test_acc"][0], ann_logs["20-shot_pre-test_acc"][0], ann_logs["50-shot_pre-test_acc"][0]))
print("|     | [7-9]  | -                     |{:0.2f}%|{:0.2f}%|{:0.2f}% |{:0.2f}% |{:0.2f}% |".format(ann_logs["1-shot_test_acc"][0], ann_logs["5-shot_test_acc"][0], ann_logs["10-shot_test_acc"][0], ann_logs["20-shot_test_acc"][0], ann_logs["50-shot_test_acc"][0]))
print("| SNN | [0-6]  | {:0.2f}% / {:0.2f}%   |{:0.2f}%|{:0.2f}%|{:0.2f}% |{:0.2f}% |{:0.2f}% |".format(max(snn_logs["pre-train_acc"]),max(snn_logs["pre-test_acc"]), snn_logs["1-shot_pre-test_acc"][0], snn_logs["5-shot_pre-test_acc"][0], snn_logs["10-shot_pre-test_acc"][0], snn_logs["20-shot_pre-test_acc"][0], snn_logs["50-shot_pre-test_acc"][0]))
print("|     | [7-9]  | -                     |{:0.2f}%|{:0.2f}%|{:0.2f}% |{:0.2f}% |{:0.2f}% |".format(snn_logs["1-shot_test_acc"][0], snn_logs["5-shot_test_acc"][0], snn_logs["10-shot_test_acc"][0], snn_logs["20-shot_test_acc"][0], snn_logs["50-shot_test_acc"][0]))


## Freezing 2 first convs - lr=1e-3 -> 1e-2 - batch_ratio=8 - pre_train_epochs=50 - no dropout - reset
|     | Labels | Pre-training on [0-6] | 1 shot | 5 shot | 10 shot | 20 shot | 50 shot |
| :-: | :----: | :-------------------: | :----: | :----: | :-----: | :-----: | :-----: |
| ANN | [0-6]  | 99.97% / 99.60%   |4.19%|0.00%|0.00% |0.00% |0.00% |
|     | [7-9]  | -                     |0.00%|33.33%|63.33% |85.00% |84.67% |
| SNN | [0-6]  | 99.98% / 99.59%   |1.73%|11.55%|0.03% |2.32% |0.00% |
|     | [7-9]  | -                     |100.00%|46.67%|83.33% |95.00% |92.67% |


## Freezing 2 first convs - lr=1e-3 -> 1e-2 - batch_ratio=8 - pre_train_epochs=50 - no dropout - reset
|     | Labels | Pre-training on [0-6] | 1 shot | 5 shot | 10 shot | 20 shot | 50 shot |
| :-: | :----: | :-------------------: | :----: | :----: | :-----: | :-----: | :-----: |
| ANN | [0-6]  | 99.97% / 99.60%   |4.19%|0.00%|0.00% |0.00% |0.00% |
|     | [7-9]  | -                     |0.00%|33.33%|63.33% |85.00% |84.67% |
| SNN | [0-6]  | 99.98% / 99.59%   |1.73%|11.55%|0.03% |2.32% |0.00% |
|     | [7-9]  | -                     |100.00%|46.67%|83.33% |95.00% |92.67% |